libarchive/archive_read_support_format_tar.c

```diff
@@ -3395,7 +3405,7 @@ readline(struct archive_read *a, struct tar *tar, const char **start,
 	tar_flush_unconsumed(a, unconsumed);
 
 	t = __archive_read_ahead(a, 1, &bytes_read);
-	if (bytes_read <= 0)
+	if (bytes_read <= 0 || t == NULL)
```

Comment:
The `bytes_read <= 0` clause here is technically redundant, but I've left it for now since it does no harm.